### DISCUSSION TITLE
How do I access Cookies and Request Object inside Route Middlewares?

### DISCUSSION THREAD

**Discussion by siamekanto19** [nuxt/nuxt#17625] [Questions]
The Route Middleware in Nuxt 3 seems like to take only two parameters `to` and `from`. But I need to check for the Cookie in the request object and block/allow Route navigation based on the presense of the cookie. This should be pretty straight forward, but I can't seem to find what I need to do to make this happen.

**Comment by viii47** [comment]
You can check the cookie with [useCookie](https://v3.nuxtjs.org/api/composables/use-cookie) inside the callback.
